src/disco/addonManager.js

```diff
@@ -36,3 +42,23 @@ export function uninstall(guid, {_mozAddonManager = window.navigator.mozAddonMan
       return;
     });
 }
+
+export function addChangeListeners(callback, mozAddonManager) {
+  function handleChangeEvent(e) {
+    const { id, type, needsRestart } = e;
+    log.info('Event received', {type, id, needsRestart});
+    if (globalEventStatusMap.hasOwnProperty(type)) {
```

Comment:
I feel this should be a `Map` if it has `Map` in the name. Unfortunately it looks like `Map` needs to be initialized with an array of key/value pairs but that's probably still nicer than using a plain object.